components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/jms/factory/JMSConnectionFactory.java

```diff
@@ -190,6 +190,7 @@ public Connection createConnection() {
                 	connection = ((TopicConnectionFactory) (this.connectionFactory))
                             .createTopicConnection();
                 }
+	            return connection;
```

Comment:
Thanks a lot for pointing this out. This is fixed with PR https://github.com/wso2/carbon-mediation/pull/459